### PR TITLE
Hawq 546

### PR DESCRIPTION
**Pull request by sansanichfb**

### CODE REVIEW

**Comment by shivzone** [comment]
Can you please add a brief title/note/description about this change

**Comment by sansanichfb** [comment]
@shivzone please take a look at JIRA description - https://issues.apache.org/jira/browse/HAWQ-546

File: src/bin/psql/describe.c [hunk 1]
```diff
@@ -4213,3 +4227,100 @@ printACLColumn(PQExpBuffer buf, const char *colname)
...
+
+
+static void
+parsePxfPattern(const char *user_pattern, char **pattern)
+{
+	strtok(strdup(user_pattern), ".");
+	*pattern = strtok(NULL, "/0");
+}
+
+static bool
```
Comment (shivzone) [inline src/bin/psql/describe.c]: add comments describing the function

File: src/bin/psql/describe.c [hunk 2]
```diff
@@ -4213,3 +4227,100 @@ printACLColumn(PQExpBuffer buf, const char *colname)
 						  "pg_catalog.array_to_string(%s, '\\n') AS \"%s\"",
 						  colname, gettext_noop("Access privileges"));
 }
+
+
+static void
```
Comment (shivzone) [inline src/bin/psql/describe.c]: function level comment required

File: src/bin/psql/describe.c [hunk 3]
```diff
@@ -1152,6 +1157,15 @@ describeTableDetails(const char *pattern, bool verbose, bool showSystem)
 	PGresult   *res;
 	int			i;
 
+	//Hive hook in this method
```
Comment (shivzone) [inline src/bin/psql/describe.c]: pattern null check required

File: src/bin/psql/describe.c [hunk 4]
```diff
@@ -4213,3 +4227,100 @@ printACLColumn(PQExpBuffer buf, const char *colname)
 						  "pg_catalog.array_to_string(%s, '\\n') AS \"%s\"",
 						  colname, gettext_noop("Access privileges"));
 }
+
+
+static void
+parsePxfPattern(const char *user_pattern, char **pattern)
+{
+	strtok(strdup(user_pattern), ".");
```
Comment (shivzone) [inline src/bin/psql/describe.c]: will this work if the first strtok doesn't find the token ?

File: src/bin/psql/describe.c (hunk 3 above)
Comment (sansanichfb) [inline src/bin/psql/describe.c]: Added.

File: src/bin/psql/describe.c (hunk 2 above)
Comment (sansanichfb) [inline src/bin/psql/describe.c]: Added comment

File: src/bin/psql/describe.c (hunk 4 above)
Comment (sansanichfb) [inline src/bin/psql/describe.c]: Good catch, added validation.

File: src/bin/psql/describe.c (hunk 1 above)
Comment (sansanichfb) [inline src/bin/psql/describe.c]: Added

File: src/bin/psql/describe.c [hunk 5]
```diff
@@ -4213,3 +4236,113 @@ printACLColumn(PQExpBuffer buf, const char *colname)
...
+
+	/* Header */
+	headers[0] = gettext_noop("Column");
+	headers[1] = gettext_noop("Type");
+	cols = 2;
+
+	char *previous_path = NULL;
+	char *previous_itemname = NULL;
+
+
```
Comment (shivzone) [inline src/bin/psql/describe.c]: extra newline

**Comment by shivzone** [comment]
+1

**Comment by realdawn** [comment]
Hi sansanichfb, it seems that your merge has conflicts and can't pass through the Jenkins test, neither do  the  pull requests based on your merge.  Please fix your merge. 
See https://builds.apache.org/job/HAWQ-build-pullrequest/535/console for more information.

**Pull request closed without merging**